### ISSUE TITLE
Bi-LSTM model setting(LSTM branch)

### ISSUE THREAD

**Issue by skkulsc** [skkulsc/text#3]
Classification_LSTM Model setting --- 2018-12-01 10:06:20.862306

for_testing = [ False ]

trainingDataPath = [ ../data/merged_dataset/train.txt ]

validationDataPath = [ ../data/merged_dataset/valid.txt ]

testingDataPath = [ ../data/merged_dataset/test.txt ]

wordDicPath = [ ../data/word_embedding_100000/wordDictionary_100000.pkl ] - vocab_size = [ 100000]

Model Training Epochs = [   300]

batch_size = [ 512] - update_size = [ 512] - training_step = [   1]

sequence_length = [ 300] - category_size = [ 6] - embedding_size = [ 512], embedding_trainable = [True]

train_X.shape = [ (520601, 300) ] - valid_X.shape = [ (142510, 300) ]

embeddingWeight.shape = [ (100000, 512) ]

lstm_hidden_sizes = [256, 128] - fc_output = [ 128] 

lr = [ 0.00100] - lr_decay = [ 0.97000] - dropout_keep_prob = [0.50]

train_X.shape = [ (520601, 300) ] - valid_X.shape = [ (142510, 300) ]

save_folder = [ ./Classification_LSTM/2018-12-01__10:03:30 ]

save_check_point_name = [ ./Classification_LSTM/2018-12-01__10:03:30/model/Classification_LSTM_model ]

check_point_file = [ None ]

-------------------------------------------------------------------------------------------

 ********************************** Classification_LSTM_model structure **********************************
< sequence_inputs > : 
Tensor("sequence_inputs:0", shape=(?, 300), dtype=int32)

< category_inputs > : 
Tensor("category_inputs:0", shape=(?,), dtype=int32)

< one_hot_category_inputs > : 
Tensor("one_hot:0", shape=(?, 6), dtype=float32)

< embeddings > : 
<tf.Variable 'embeddings:0' shape=(100000, 512) dtype=float32_ref>

< sequence_emb > : 
Tensor("sequence_emb/Identity:0", shape=(?, 300, 512), dtype=float32, device=/device:CPU:0)

< lstm_fw_cells[0] > : 
<tensorflow.python.ops.rnn_cell_impl.LSTMCell object at 0x7fca0d845278>

< lstm_fw_drops[0] > : 
<tensorflow.python.ops.rnn_cell_impl.DropoutWrapper object at 0x7fca0d845908>

 < fw_state > : 
(LSTMStateTuple(c=<tf.Tensor 'Bi-LSTM/BiLSTM/cell_0/bidirectional_rnn/fw/fw/while/Exit_3:0' shape=(?, 256) dtype=float32>, h=<tf.Tensor 'Bi-LSTM/BiLSTM/cell_0/bidirectional_rnn/fw/fw/while/Exit_4:0' shape=(?, 256) dtype=float32>), LSTMStateTuple(c=<tf.Tensor 'Bi-LSTM/BiLSTM/cell_1/bidirectional_rnn/fw/fw/while/Exit_3:0' shape=(?, 128) dtype=float32>, h=<tf.Tensor 'Bi-LSTM/BiLSTM/cell_1/bidirectional_rnn/fw/fw/while/Exit_4:0' shape=(?, 128) dtype=float32>))

 < bw_state > : 
(LSTMStateTuple(c=<tf.Tensor 'Bi-LSTM/BiLSTM/cell_0/bidirectional_rnn/bw/bw/while/Exit_3:0' shape=(?, 256) dtype=float32>, h=<tf.Tensor 'Bi-LSTM/BiLSTM/cell_0/bidirectional_rnn/bw/bw/while/Exit_4:0' shape=(?, 256) dtype=float32>), LSTMStateTuple(c=<tf.Tensor 'Bi-LSTM/BiLSTM/cell_1/bidirectional_rnn/bw/bw/while/Exit_3:0' shape=(?, 128) dtype=float32>, h=<tf.Tensor 'Bi-LSTM/BiLSTM/cell_1/bidirectional_rnn/bw/bw/while/Exit_4:0' shape=(?, 128) dtype=float32>))

 < concat_states > : 
Tensor("Bi-LSTM/BiLSTM_states_concat:0", shape=(?, 256), dtype=float32)

< lstm_fc_w > : 
<tf.Variable 'Bi-LSTM-fc/lstm_fc_w:0' shape=(256, 128) dtype=float32_ref>

< lstm_fc_b > : 
<tf.Variable 'Bi-LSTM-fc/lstm_fc_b:0' shape=(128,) dtype=float32_ref>

< lstm_fc_output > : 
Tensor("Bi-LSTM-fc/lstm_fc_output:0", shape=(?, 128), dtype=float32)

< W_output > : 
<tf.Variable 'output/W_output:0' shape=(128, 6) dtype=float32_ref>

< b_output > : 
<tf.Variable 'output/b_output:0' shape=(6,) dtype=float32_ref>

< scores > : 
Tensor("output/scores:0", shape=(?, 6), dtype=float32)

< loss > : 
Tensor("loss/Mean:0", shape=(), dtype=float32)

< predictions > : 
Tensor("output/predictions:0", shape=(?,), dtype=int64)

< correct_predictions > : 
Tensor("accuracy/Equal:0", shape=(?,), dtype=bool)

< accuracy > : 
Tensor("accuracy/accuracy:0", shape=(), dtype=float32)

< loss_summary > : 
Tensor("summary/train_loss:0", shape=(), dtype=string)

< acc_summary > : 
Tensor("summary/train_accuracy:0", shape=(), dtype=string)

< lr_summary > : 
Tensor("summary/learning_rate:0", shape=(), dtype=string)

< train_summary_op > : 
Tensor("summary/Merge/MergeSummary:0", shape=(), dtype=string)

< test_summary_op > : 
Tensor("summary/Merge_1/MergeSummary:0", shape=(), dtype=string)

< gradients > : 
[<tensorflow.python.framework.ops.IndexedSlices object at 0x7fc9cbbb4f28>, <tf.Tensor 'optimizer/clip_by_global_norm/optimizer/clip_by_global_norm/_1:0' shape=(768, 1024) dtype=float32>, <tf.Tensor 'optimizer/clip_by_global_norm/optimizer/clip_by_global_norm/_2:0' shape=(1024,) dtype=float32>, <tf.Tensor 'optimizer/clip_by_global_norm/optimizer/clip_by_global_norm/_3:0' shape=(768, 1024) dtype=float32>, <tf.Tensor 'optimizer/clip_by_global_norm/optimizer/clip_by_global_norm/_4:0' shape=(1024,) dtype=float32>, <tf.Tensor 'optimizer/clip_by_global_norm/optimizer/clip_by_global_norm/_5:0' shape=(640, 512) dtype=float32>, <tf.Tensor 'optimizer/clip_by_global_norm/optimizer/clip_by_global_norm/_6:0' shape=(512,) dtype=float32>, <tf.Tensor 'optimizer/clip_by_global_norm/optimizer/clip_by_global_norm/_7:0' shape=(640, 512) dtype=float32>, <tf.Tensor 'optimizer/clip_by_global_norm/optimizer/clip_by_global_norm/_8:0' shape=(512,) dtype=float32>, <tf.Tensor 'optimizer/clip_by_global_norm/optimizer/clip_by_global_norm/_9:0' shape=(256, 128) dtype=float32>, <tf.Tensor 'optimizer/clip_by_global_norm/optimizer/clip_by_global_norm/_10:0' shape=(128,) dtype=float32>, <tf.Tensor 'optimizer/clip_by_global_norm/optimizer/clip_by_global_norm/_11:0' shape=(128, 6) dtype=float32>, <tf.Tensor 'optimizer/clip_by_global_norm/optimizer/clip_by_global_norm/_12:0' shape=(6,) dtype=float32>]

< optimizer > : 
<tensorflow.python.training.adam.AdamOptimizer object at 0x7fc9cca5ceb8>

< train_op > : 
name: "optimizer/Adam"
op: "NoOp"
input: "^optimizer/Adam/update_embeddings/group_deps"
input: "^optimizer/Adam/update_BiLSTM/cell_0/bidirectional_rnn/fw/LSTM-forward-hiddenSize-256/kernel/ApplyAdam"
input: "^optimizer/Adam/update_BiLSTM/cell_0/bidirectional_rnn/fw/LSTM-forward-hiddenSize-256/bias/ApplyAdam"
input: "^optimizer/Adam/update_BiLSTM/cell_0/bidirectional_rnn/bw/LSTM-backward-hiddenSize-256/kernel/ApplyAdam"
input: "^optimizer/Adam/update_BiLSTM/cell_0/bidirectional_rnn/bw/LSTM-backward-hiddenSize-256/bias/ApplyAdam"
input: "^optimizer/Adam/update_BiLSTM/cell_1/bidirectional_rnn/fw/LSTM-forward-hiddenSize-128/kernel/ApplyAdam"
input: "^optimizer/Adam/update_BiLSTM/cell_1/bidirectional_rnn/fw/LSTM-forward-hiddenSize-128/bias/ApplyAdam"
input: "^optimizer/Adam/update_BiLSTM/cell_1/bidirectional_rnn/bw/LSTM-backward-hiddenSize-128/kernel/ApplyAdam"
input: "^optimizer/Adam/update_BiLSTM/cell_1/bidirectional_rnn/bw/LSTM-backward-hiddenSize-128/bias/ApplyAdam"
input: "^optimizer/Adam/update_Bi-LSTM-fc/lstm_fc_w/ApplyAdam"
input: "^optimizer/Adam/update_Bi-LSTM-fc/lstm_fc_b/ApplyAdam"
input: "^optimizer/Adam/update_output/W_output/ApplyAdam"
input: "^optimizer/Adam/update_output/b_output/ApplyAdam"
input: "^optimizer/Adam/Assign"
input: "^optimizer/Adam/Assign_1"

# train accuracy, loss summary (1 epoch 당 약 1000개의 batch)

<img width="677" alt="lstm_train_summary" src="https://user-images.githubusercontent.com/37528988/49327915-ec26a080-f5aa-11e8-85ce-2a45dc5798ea.png">

# test accuracy, loss summary (1 epoch 당 약 280개의 batch)

<img width="673" alt="lstm_test_summary" src="https://user-images.githubusercontent.com/37528988/49327919-02ccf780-f5ab-11e8-80eb-535d2f74f66f.png">
